registry/auth/openshift/accessController.go

```diff
@@ -0,0 +1,148 @@
+package openshift
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	ctxu "github.com/docker/distribution/context"
+	"github.com/docker/distribution/registry/auth"
+	"golang.org/x/net/context"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+)
+
+type accessController struct {
+	openshiftAddr       string
+	openshiftApiVersion string
+	client              *http.Client
+}
+
+type openshiftToken struct {
+	namespace   string
+	imageName   string
+	verb        string
+	bearerToken string
+}
+
+var _ auth.AccessController = &accessController{}
+
+func (ac *accessController) verifyOpenShiftToken(token openshiftToken) error {
+	url := fmt.Sprintf("https://%s/subjectAccessReviews?namespace=%s", ac.openshiftAddr, token.namespace)
```

Comment:
We should probably switch to using the Client from origin and use SubjectAccessReviewNamespacer, etc. But that can wait.